### PR TITLE
feature/305 - Update to pytest 8

### PR DESCRIPTION
**Pull request by sasha-dresden**
Issue: #305

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
 Sphinx==4.3.1
-pytest==7.4.4
+pytest
 coverage==6.2
 pytest-cov
 pylint==2.13.9
 jsonschema==3.2.0
-json-schema-for-humans==0.39.1
```
Comment (mjtravers) [inline requirements.txt:7]: Let's pin pytest, json-schema-for-humans, and pytest-cov to the latest appropriate version if we can so we can control upgrades.

Also, it seems like a number of the packages in requirements.txt should be in requirements-test.txt. We are concatinating the two files in .circleci/config.yml

**Review by mjtravers — CHANGES_REQUESTED** [review]
See inline comment

File: requirements.txt (hunk 1 above)
Comment (sasha-dresden) [reply requirements.txt:7]: Cleaned it up so only test related packages should be requirements-test.txt and no test related ones are in requirements.txt. Without any duplication.
Also removed pinning the versions.

Comment (mjtravers) [reply requirements.txt:7]: Sorry for the miscommunication. We actually want the packages pinned to a version so we don't get surprised with any breaking changes. The request is to pin the unpinned packages.

**Review by mjtravers — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (sasha-dresden) [reply requirements.txt:7]: Ah sorry I read that backwards. Got it fixed so now every package has a version number

**Review by mjtravers — APPROVED** [review]

**Pull request merged**